lua/core/menu/tape.lua

```diff
@@ -50,48 +50,30 @@ local function tape_exists(index)
 end
 
 local function read_tape_index()
-  local f = io.open(_path.tape..'index.txt','r')
-  if f ~= nil then
-    local a = tonumber(f:read("*line"))
-    m.fileindex = a or 0
-    f:close()
-  else
-    m.fileindex = 0
-  end
-  while tape_exists(m.fileindex) do
-    m.fileindex = m.fileindex+1
-  end
-end
-
-local function write_tape_index_v(v)
-  local f = io.open(_path.tape..'index.txt','w')
-  if f == nil then
-    os.execute("mkdir -p ".._path.tape)
-    f = io.open(_path.tape..'index.txt','w')
+  tape = util.os_capture("ls ".._path.audio.."tape/", true)
```

Comment:
could you add a check here that if the tape folder isn't found, the folder is automatically created?

otherwise looks fantastic! much more robust. thank you.

Reply:
It's safe to run `mkdir -p` even if the folder exists, so I popped it in that way. I can wrap it in a conditional if that's preferred, though.